ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -40,22 +40,25 @@
 
   private static final Map<Class, Function<Integer, String>> typeConverters
       = ImmutableMap.<Class, Function<Integer, String>>builder()
-      .put(int.class, index -> typeConversionCode("Integer", index))
-      .put(Integer.class, index -> typeConversionCode("Integer", index))
-      .put(long.class, index -> typeConversionCode("Long", index))
-      .put(Long.class, index -> typeConversionCode("Long", index))
-      .put(double.class, index -> typeConversionCode("Double", index))
-      .put(Double.class, index -> typeConversionCode("Double", index))
-      .put(boolean.class, index -> typeConversionCode("Boolean", index))
-      .put(Boolean.class, index -> typeConversionCode("Boolean", index))
-      .put(String.class, index -> typeConversionCode("String", index))
-      .put(Map.class, index -> typeConversionCode("Map", index))
-      .put(List.class, index -> typeConversionCode("List", index))
+      .put(int.class, index -> typeConversionCode("Integer", index, true))
+      .put(Integer.class, index -> typeConversionCode("Integer", index, false))
+      .put(long.class, index -> typeConversionCode("Long", index, true))
+      .put(Long.class, index -> typeConversionCode("Long", index, false))
+      .put(double.class, index -> typeConversionCode("Double", index, true))
+      .put(Double.class, index -> typeConversionCode("Double", index, false))
+      .put(boolean.class, index -> typeConversionCode("Boolean", index, true))
+      .put(Boolean.class, index -> typeConversionCode("Boolean", index, false))
+      .put(String.class, index -> typeConversionCode("String", index, false))
+      .put(Map.class, index -> typeConversionCode("Map", index, false))
+      .put(List.class, index -> typeConversionCode("List", index, false))
       .build();
 
   // Templates used to generate the UDF code
   private static final String genericTemplate =
       "#TYPE arg#INDEX;\n"
+          + "if(args[#INDEX] == null && #IS_PRIMITIVE)\n"
+          + "throw new KsqlFunctionException(\"Can't coerce argument at index #INDEX from "
```

Comment:
The comment below which has an example of the generated code will have to be updated to include the new check for a nullable type.